packages/flutter/test/widgets/scroll_simulation_test.dart

```diff
@@ -23,4 +25,98 @@ void main() {
     checkInitialConditions(75.0, 614.2093);
     checkInitialConditions(5469.0, 182.114534);
   });
+
+  test('ClampingScrollSimulation only decelerates, never speeds up', () {
```

Comment:
Can we add a test that the velocity eventually goes to zero?

Reply:
Sure, will do.

I think that follows as a consequence of being ballistic/restartable: if the simulation still has significant velocity in the last moments before the end (i.e., if the limit as time goes to the ending time isn't zero), then restarting it at that point would mean it goes on for longer than the few moments that were otherwise remaining.

But it would still be sensible to have a test for it directly.

Reply:
Added.